woocommerce/woocommerce-android
# Issue/11369 onboarding card UI update

Closes: #11369


### Description

This PR contains a refactor to adapt existing onboarding view to the new dynamic dashboard logic: 
- Adds a new `DashboardOnboardingViewModel`. Couldn't reuse the existing `StoreOnboardingViewModel` as there were many changes that made reusing it unfeasible. 
- Adds a new composable view `DashboardOnboardingCard`
- Extract logic out of existing `StoreOnboardingViewModel` to be reusable (there's further "house cleaning" that can be done in this ViewModel once we activate all the new dynamic dashboard code)

NOTE: Notice the main CTA of the onboarding card is messed up. This is because the `DashboardWidgetAction` model used in `WidgetCard` only accepts a `@StringRes val titleResource: Int,` as title. So I'll need to refactor `DashboardWidgetAction` to use `val title: String` instead so that we are able to use string resources that accept parameters. I'll handle that refactor in a different PR as that'd add a lot of noise to these changes.

### Testing instructions

1. Enable dynamic dashboard feature flag
2. Check that the onboarding list is displayed correctly
3. When tapping on `Hide setup list` notice how you navigate to widget editor screen
4. Tap on `Share feedback` and check feedback survey is loaded
5. Tap on any of the task from the list and check you are taken to the correct screen
6. Swipe to refresh and check the skeletons are shown while list is fetched

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/f5d7f330-a492-4b92-a3cf-0187a9b68b85
